app/models/admin_profile.rb

```diff
@@ -2,4 +2,13 @@
 
 class AdminProfile < ApplicationRecord
   belongs_to :user
+
+  include Discard::Model
+  default_scope -> { kept }
+
+  scope :kept, -> { undiscarded.joins(:user).merge(User.kept) }
```

Comment:
Interesting. How do default scopes work with one-to-one / many-to-one relations in Rails? 

If we have a user, who has an admin profile, and then we discard the admin profile, will the user check for it being admin work? 

And I guess, if we have a school which has many teacher profiles, and we discard some of them, will the school only get the non-discarded when we ask for the list of its teachers?

(and I hope it propagates through `through` keyword?) 

Just something to test and think about, maybe even to write a unit test about. Pretty sure you are thinking about it already :)

Reply:
1. Q: If we have a user, who has an admin profile, and then we discard the admin profile, will the user check for it being admin work?
2. A: if we would discard User profile - it would hide/soft delete the corresponding profile. Should we consider the opposite scenario i.e if profile is soft deleted - the corresponding user should be hidden as well - Good question - what do you think should happen @kicferk1 
3. Q And I guess, if we have a school which has many teacher profiles, and we discard some of them, will the school only get the non-discarded when we ask for the list of its teachers?
4. A: Given that the default scope is for kept records - we would see only the non discarded ones - and having a test will prove it
5. Q: (and I hope it propagates through through keyword?) - 
6. A: The unit test will show us the answer 

Good points - if you think of any other scenarios - let me know